force-app/main/default/classes/CourseRegistrationHandlerTest.cls

```diff
@@ -0,0 +1,504 @@
+@isTest
+public with sharing class CourseRegistrationHandlerTest {
+    @TestSetup
+    static void makeData() {
+        TestDataFactory.insertEncryptionKey();
+    }
+
+    @isTest
+    private static void testInsertDeleteUndelete() {
```

Comment:
Lag én test for insert, én test for delete og én test for undelete.
I f.eks. undelete trenger du ikke å teste at insert og delete fungerer, fordi det skjer i `testInsert` og i `testDelete`